src/south/SouthConnector.class.js

```diff
@@ -0,0 +1,401 @@
+
+          this.lastCompletedAt[scanMode] = lastCompletedAt || defaultLastCompletedAt
+          this.queryParts[scanMode] = parseInt(queryPart || '0', 10)
+          this.ignoredReadsCounters[scanMode] = 0
+          this.currentlyOnScan[scanMode] = 0
+        }
+      })
+    } else {
+      this.logger.error(`Scan mode or scan groups for South "${this.settings.name}" are not defined.`
+          + 'This South connector will not work.')
+      this.scanGroups = []
+    }
+
+    if (!supportListen && !supportLastPoint && !supportFile && !supportHistory) {
+      this.logger.error(`${this.constructor.name} should support at least 1 operation mode.`)
+    }
+    if (supportListen && typeof this.listen !== 'function') {
+      this.logger.error(`${this.constructor.name} should implement the listen() method.`)
+    }
+    if (supportLastPoint && typeof this.lastPointQuery !== 'function') {
+      this.logger.error(`${this.constructor.name} should implement the lastPointQuery() method.`)
+    }
+    if (supportFile && typeof this.fileQuery !== 'function') {
+      this.logger.error(`${this.constructor.name} should implement the fileQuery() method.`)
+    }
+    if (supportHistory && typeof this.historyQuery !== 'function') {
+      this.logger.error(`${this.constructor.name} should implement the historyQuery() method.`)
+    }
+
+    this.statusService.updateStatusDataStream({
+      'Number of values since OIBus has started': this.handlesPoints ? 0 : undefined,
+      'Number of files since OIBus has started': this.handlesFiles ? 0 : undefined,
+    })
+    this.logger.info(`South connector "${name}" (${id}) of type ${protocol} started.`)
+  }
+
+  /**
+   * Method called by Engine to initialize a South connector. This method can be surcharged in the
+   * South connector implementation to allow connection to a third party application for example.
+   * @returns {Promise<void>} - The result promise
+   */
+  async connect() {
+    this.connected = true
+    this.statusService.updateStatusDataStream({ 'Connected at': new Date().toISOString() })
+  }
+
+  /**
+   * Method used to manage history queries of a South connector. If needed, a query interval will be split
+   * @param {String} scanMode - The scan mode to activate
+   * @param {Date} historyStartTime - The start time of the history query
+   * @param {Date} historyEndTime - The end time of the history query
+   *                -set to now in OIBus Engine data stream
+   *                -set to a specific time in HistoryQueryEngine
+   * @returns {Promise<void>} - The result promise
+   */
+  async historyQueryHandler(scanMode, historyStartTime, historyEndTime) {
+    // maxReadInterval will divide a huge request (for example 1 year of data) into smaller
+    // requests. For example only one hour if maxReadInterval is 3600 (in s)
+    const intervals = generateIntervals(historyStartTime, historyEndTime, this.maxReadInterval)
```

Comment:
It may not be correct. If I understand correctly the logic for intervals generation the first interval endTime will be equal with the second interval startTime. The problem is that Javascript handles only milliseconds, but databases and OPC can handle smaller time intervals which can cause duplicate values.
E.g. 
first ineterval endTime: "2022-09-16 11:11:11.111"
second ineterval startTime: "2022-09-16 11:11:11.111"
DB/OPC timestamp: "2022-09-16 11:11:11.11111"
In this case, I think, the value will be present in both intervals.

Reply:
I agree, however I don't see how to manage this edge case for now. 
I think this case was happening before the refactor, no ? 
Should we put an issue ? I don't think we have a way to avoid that since it's dependent of the node-opcua library.

Reply:
Probably it would be a good idea to make an issue for this. At some point we had something like: `second interval startTime = first interval endTime + 1 (millisecond)`, but I'm not 100% sure that it would properly cover all edge cases.